internal/scheduler/scheduler.go

```diff
@@ -200,17 +207,20 @@ func (s *Scheduler) cycle(ctx context.Context, updateAll bool, leaderToken Leade
 	}
 	events = append(events, expirationEvents...)
 
-	// Schedule jobs.
-	overallSchedulerResult, err := s.schedulingAlgo.Schedule(ctx, txn, s.jobDb)
-	if err != nil {
-		return err
-	}
+	if s.clock.Now().Sub(s.previousSchedulingRoundEnd) > s.schedulingPeriod {
```

Comment:
let's log (maybe at debug) about whether we decided to schedule or not and why.  It'll come in useful if we ever end up not scheduling and we can't work out why.